castle

whuan20011 opened the pull request:
Algorithm: bfs
do bfs in a graph by using while loop and queue to find out the existing adjcent neighboors, then follow the loop
write a function to find out neighboors(watch out if, if, if, if, not if, elif, elif, elif)
watch out the row and column in a list of list

whuan20011 committed:
castle
Algorithm: bfs
do bfs in a graph by using while loop and queue to find out the existing adjcent neighboors, then follow the loop
write a function to find out neighboors(watch out if, if, if, if, not if, elif, elif, elif)
watch out the row and column in a list of list

smy20011 commented on castle/castle.py:
```diff
@@ -0,0 +1,113 @@
+    for i in range(n):
+        for j in range(m):
+            cur = maps[i][j]
+            if j - 1 >= 0 and maps[i][j - 1] != cur:
```
You can reuse the get_neighboors function here.

smy20011 commented on castle/castle.py:
```diff
@@ -0,0 +1,113 @@
+                            queue.append(nei)
+                room += 1
+                room_size[room] = len(visited)
+                biggest_size = max(biggest_size, len(visited))
```
biggest_size is not used.

smy20011 commented on castle/castle.py:
```diff
@@ -0,0 +1,113 @@
+            south = max(wall[0], south)
+    print >> fout, room
+    print >>fout, biggest_size
+    print >>fout,  bigger_size
```
All print statements should formatted same.

whuan20011 committed:
castle
Algorithm: bfs(by using while loop and queue)
function bigger_size got changed by using find_neighboors
add west_south to make the print easier
